Automattic/jetpack
# Add media validator for image picker

This change introduces a media validator that helps in to set the maximum filesize and allowed types based on the enabled connections. The UI part is still being designed by the design team, so for now I just added a few notices so we can see what is happening.

#### Other information:


#### Jetpack product discussion



1203425056138987-as-1203411993625377

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






`testimage1`: www.cloudup.com/cXrFq_uE782
`testimage2`: www.cloudup.com/cJphGqezDB1

* Create a Facebook and a Twitter Connection, and change Twitter maxImage size to 2, and Facebook to 1 in the restrictions.
* Go to the editor, you should see the current maximum filesize and typed below the image picker. 1Mb if both services is enabled, 2Mb if only Twitter.
* Try to upload  `testimage2`, it should not get selected and a Notice appears, which you can close. It should also close by uploading a valid image. Upload `testimage1`, it should work.
* Disable Facebook, so only Twitter is enabled. Upload `testimage2` it should work. Turn on Facebook again, it should unselect the image and bring up the notice.
